### PR TITLE
Updates Strong-Arm implants to be cooldown based, deal entirely lethal damage (but more of it)

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Strong-Arm implants utilize a five second cooldown between activation (this is per implant, not for both). As a result, the implant does a larger chunk of damage in one blow. It does not do stamina damage anymore.

If you strike a mob with either the MOB_BEAST or MOB_SPECIAL biotypes, you inflict additional damage (40 extra).

You can fit both a toolset implant and a muscle implant into the same arm.

## Why It's Good For The Game

Strong-arms were really weird implants in that they were not locked behind any research progression, but instead could be purchased from cargo. However, they were still not exactly amazing to have, so they only really get purchased by bored medical doctors, since the implant was a medical departmental order option (for some reason). I tried giving them a little extra oomph a while ago, but I never felt like they were in a particularly good space even after that update.

Borrowing some ideas from an older PR, I've introduced a cooldown to allow for them to work alongside other unarmed features, but still not allow them to stack with other effects. It makes them exceptionally good as opening attacks, but not to the point of being intrusive. They were also a method of infinite stamina damage, so that has been resolved as well.

The mob bonus damage makes them a little fun option for miners, really, that's why I added it. I thought it would be a cute feature, and maybe incentivize a few to get augmented. I don't think they will, since augments aren't super useful for miners, but still.

The pairing with toolsets is because they're not quite as useful generally as the toolsets, and since they are a weaker arm implant in the progression, I don't feel it is too extreme to have both, because overall most people will gravitate towards toolsets if they had to choose.

## Changelog
:cl:
balance: Strong-arm implants go on cooldown when activated, but deal more overall lethal damage on a single blow.
balance: The strong-arm implant does additional damage to fauna and some other dangerous mobs.
qol: You can pair toolset arm implants with the strong-arm implants.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
I understand the reasoning but it makes me sad every time we move cargo content into lathes instead. Are you absolutely certain it needs the research lock?

**Comment by necromanceranne** [comment]
_Shrug_ I don't think it should have been in cargo to begin with. It's the only cybernetic in cargo, but because it is a departmental crate, it is locked from being conveniently accessible to the other half of the departments that deal with cybernetics. If I moved it to a science crate, the same goes for medical.

When I say 'weaker', I mean 'weaker than toolsets', not 'weak overall'. I wouldn't say no to one, but if I had to choose, I know what I would over either implant.

**Comment by necromanceranne** [comment]
Oh, and to be entirely honest, I think having it in cargo means that most people don't actually know it exists? I think most people I speak to know what rebooters, revivers and toolsets are, but when I say 'oh yeah, but what about the strong-arm' they're like 'what the fuck is a strong-arm?'

**Comment by bob-b-b** [comment]
How would this work with the goliath dna infusion? Seems like it'd be a cool combo with the hulk arm that gives you for even more punch damage.

**Comment by Borisvanmemes** [comment]
How much extra damage do they actually do now?

**Comment by necromanceranne** [comment]
> How would this work with the goliath dna infusion? Seems like it'd be a cool combo with the hulk arm that gives you for even more punch damage.

No combination. One is an unarmed attack, the other is a weapon. I guess you could get the implant in the arm that isn't your goliath arm so you can keep punting mobs around. Kinda funny.

> How much extra damage do they actually do now?

| Limb Type | Average | Floor | Ceiling |
| --- | --- | --- | --- |
| Standard | 26 | 20 | 30 |
| Standard Fauna | 66 | 60 | 70 |
| Advanced | 28 | 20 | 36 |
| Advanced Fauna | 68 | 60 | 76 |

Note that armor does apply to these attacks.

**Comment by Dawnseer** [comment]
It isn't just a departmental. It can be ordered completely normally. And it isn't a locked crate either.

**Comment by necromanceranne** [comment]
> I feel like if you're aiming to make them a miner augment (which is very neat!) you should keep them in cargo.

I'm not sure if it's really meant to be, one hundred percent a mining implant. I just thought it would be neat to add that as a bonus feature so it scales against fauna. You know, for style points.

> It isn't just a departmental. It can be ordered completely normally. And it isn't a locked crate either.

I'm aware. 

_But uh, and this is very anecdotal so take it with a grain of salt, but I've been having trouble ordering stuff via cargo lately because I think the stock market has people unwilling to spend actual credits and only willing to fulfill departmental orders so they never lose the credits they need for stock market exchanges. Another anecdotal situation, I know, but it definitely is getting annoying that I've run into that a lot._

**Comment by necromanceranne** [comment]
Jacq, don't be ridiculous, that really isn't the reason why I moved it. I'm just grumbling about stock market junkies.

I am arguing that it is inconvenient to acquire for one of the departments that handle cybernetics due to it being in cargo as a consequence of how the departmental order system works. And it doesn't seem reasonable for one department to be favored over the other in this regard when we should allow either group access to it.

I'm not committed enough to die on this hill about it and I will change it back if you really care, but I don't think you're really doing me any favours by reducing my argument to something you can conveniently dismiss.

**Comment by Jacquerel** [comment]
It is something you wrote in the PR comments so I replied to it?
If we don't have a system for something to be able to be departmentally ordered by multiple departments that seems quite odd (and I am reasonably sure multiple people can order canisters of gas, so we probably do?) and seems reasonably easy to address without changing it to a different method of acquisition.

**Comment by carlarctg** [comment]
There's no need to have things have only one possible way to be procured. It would indeed be very nice if implants could be purchased at a markup before they were researched.

**Comment by necromanceranne** [comment]
> It is something you wrote in the PR comments so I replied to it? 

Sorry, I shouldn't be snapping, having a rough one. Felt like me making an off-hand remark was the only thing being acknowledged.

I'll look into it and make some changes that might solve the issue.

**Comment by EnterTheJake** [comment]
> > There's no need to have things have only one possible way to be procured.
> 
> No but unless there's a good reason, it's often preferable for things to have one way to be procured. Cargo has a massive problem that generally speaking things drift out of "you can buy this" into "you can research and print this" or "this is in station vending machines" until there's nothing worth spending money on which we haven't really been trying hard enough to push against.
> 
> I'll be honest if you were adding this as a new item I wouldn't even bat an eye about it being in R&D so ultimately I am not going to consider "this must remain in cargo" as a blocking requirement, it just feels like a shame.

I do agree cargo needs more goodies to be ordered, but in reality it's a massive pain in the ass having to go through the process of ordering an item and then for it to also require surgery.

**Comment by EnterTheJake** [comment]
> I feel like if you're aiming to make them a miner augment (which is very neat!) you should keep them in cargo, implants are usually one of the last techs researched on-par with weapons and mechs, and currently costs a quite lot to do. Miners are part of cargo so it'd probably be easier for them to bother QM/order those themselves rather than nag RND for one more thing beyond mining tech.

Actually, no they are not, you can get combat implants incredibly early into the shift, I don't have the exact number on hand, but defo in less than 8 minutes.

**Comment by Dawnseer** [comment]
Isn't the toolset implants even behind the combat implants now? Like you can get the no-drop implant before toolsets right now?

**Comment by ATH1909** [comment]
> They were also a method of infinite stamina damage, so that has been resolved as well.

Is this a problem? I thought that the changes to how staminacrit works were supposed to make having access to a repeatable source of stamina damage a non-issue.

**Comment by excalisword** [comment]
Does the strong-arm implant still synergize with the nullblade with this change?

**Comment by necromanceranne** [comment]
it never did. You're thinking of cybernetic limbs.

**Comment by TetrakoHateThis** [comment]
To add: the Strong-Arm implant, in spite of being a majority combat focused implant, is locked to Medical, rather than being with it's kin in Science. This was a fucking bizarre choice from the outset so I'm all for moving it out to a more sensible domain so roboticists can access the damn things.

**Comment by necromanceranne** [comment]
That's actually not a bug, that's intended. With a cooldown, you _can_ use martial arts while it is on cooldown, but the punch was always intended to be an override on all unarmed attacks.

**Comment by necromanceranne** [comment]
68747470733a2f2f632e74656e6f722e636f6d2f6641376f6c704f66524e3841414141642f6361742d736c656570792e676966

**Comment by Jacquerel** [comment]
arcane has implied hes going to investigate making this desired behaviour easier so maybe we can add it in post

File: code/modules/research/designs/medical_designs.dm [hunk 1]
```diff
@@ -403,7 +403,11 @@
 	desc = "A set of surgical tools hidden behind a concealed panel on the user's arm."
 	id = "ci-surgery"
 	build_type = PROTOLATHE | AWAY_LATHE | MECHFAB
-	materials = list (/datum/material/iron = SHEET_MATERIAL_AMOUNT*1.25, /datum/material/glass =HALF_SHEET_MATERIAL_AMOUNT * 1.5, /datum/material/silver =HALF_SHEET_MATERIAL_AMOUNT * 1.5)
+	materials = list (
+		/datum/material/iron = SHEET_MATERIAL_AMOUNT*1.25,
+		/datum/material/glass =HALF_SHEET_MATERIAL_AMOUNT * 1.5,
+		/datum/material/silver =HALF_SHEET_MATERIAL_AMOUNT * 1.5
```
Comment (Jacquerel) [inline code/modules/research/designs/medical_designs.dm:409]:
```suggestion
		/datum/material/iron = SHEET_MATERIAL_AMOUNT * 1.25,
		/datum/material/glass = HALF_SHEET_MATERIAL_AMOUNT * 1.5,
		/datum/material/silver = HALF_SHEET_MATERIAL_AMOUNT * 1.5,
```

File: code/modules/research/designs/medical_designs.dm [hunk 2]
```diff
@@ -416,7 +420,11 @@
 	desc = "A stripped-down version of engineering cyborg toolset, designed to be installed on subject's arm."
 	id = "ci-toolset"
 	build_type = PROTOLATHE | AWAY_LATHE | MECHFAB
-	materials = list (/datum/material/iron = SHEET_MATERIAL_AMOUNT*1.25, /datum/material/glass =HALF_SHEET_MATERIAL_AMOUNT * 1.5, /datum/material/silver =HALF_SHEET_MATERIAL_AMOUNT * 1.5)
+	materials = list (
+		/datum/material/iron = SHEET_MATERIAL_AMOUNT*1.25,
+		/datum/material/glass =HALF_SHEET_MATERIAL_AMOUNT * 1.5,
+		/datum/material/silver =HALF_SHEET_MATERIAL_AMOUNT * 1.5
```
Comment (Jacquerel) [inline code/modules/research/designs/medical_designs.dm:426]:
```suggestion
		/datum/material/iron = SHEET_MATERIAL_AMOUNT * 1.25,
		/datum/material/glass = HALF_SHEET_MATERIAL_AMOUNT * 1.5,
		/datum/material/silver = HALF_SHEET_MATERIAL_AMOUNT * 1.5,
```

File: code/modules/surgery/organs/internal/cyberimp/augments_arms.dm [hunk 3]
```diff
@@ -454,13 +473,16 @@
...
+		potential_damage += biotype_bonus_damage
+
 	source.do_attack_animation(target, ATTACK_EFFECT_SMASH)
 	playsound(living_target.loc, 'sound/weapons/punch1.ogg', 25, TRUE, -1)
 
 	var/target_zone = living_target.get_random_valid_zone(source.zone_selected)
 	var/armor_block = living_target.run_armor_check(target_zone, MELEE, armour_penetration = attacking_bodypart.unarmed_effectiveness)
-	living_target.apply_damage(potential_damage, attacking_bodypart.attack_type, target_zone, armor_block)
-	living_target.apply_damage(potential_damage*1.5, STAMINA, target_zone, armor_block)
+	living_target.apply_damage(potential_damage*2, attacking_bodypart.attack_type, target_zone, armor_block)
```
Comment (Jacquerel) [inline code/modules/surgery/organs/internal/cyberimp/augments_arms.dm:485]:
```suggestion
	living_target.apply_damage(potential_damage * 2, attacking_bodypart.attack_type, target_zone, armor_block)
```

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**